src/index.ts

```diff
@@ -77,7 +84,45 @@ async function main() {
             .reduce((a, b) => Math.max(a, b), cancelBefore.getTime());
           cancelBefore = new Date(n);
         }
-        const runningWorkflows = workflow_runs.filter(
+
+        if (disqualifying_jobs && !Array.isArray(disqualifying_jobs)) {
+          core.setFailed('Disqualifying jobs found but is not array');
+        }
+
+        const workflow_jobs = (
+          disqualifying_jobs && disqualifying_jobs.length > 0
+            ? await Promise.all(
+                workflow_runs.map(async ({ id, jobs_url }) => {
+                  const {
+                    data: { jobs },
+                  } = await octokit.request(`GET ${jobs_url}`, {
+                    owner,
+                    repo,
+                    run_id: id,
+                  });
+                  return {
+                    workflow_run_id: id,
+                    jobs: jobs.filter(
+                      ({ status, name }: any) =>
+                        status === 'in_progress' && disqualifying_jobs.includes(name),
```

Comment:
Should we use Job ID instead of Job Name? Or at least make both work like we do for workflow name/id?